app/src/androidTest/java/com/droidhats/campuscompass/views/MapFragmentTest.kt

```diff
@@ -54,12 +64,211 @@ class MapFragmentTest {
+            allOf(
+                withId(R.id.mapFragSearchBar), isDisplayed()
+            )
+        )
+        containerSearchBar.perform(click())
+
+        //Checks if text "Search" is displayed even after search bar was clicked
+        val searchText = onView(
+            allOf(
+                withId(R.id.search_src_text), withHint("Search"), isDisplayed()
+            )
+        )
+        searchText.check(matches(withHint("Search")))
+
+        //Checks if info text that prompts to enter the text in the text field is displayed
+        val infoTextView = onView(
+            allOf(
+                withId(R.id.search_info), withText("Enter Street, Address, Concordia Classroom..."),
+                isDisplayed()
+            )
+        )
+        infoTextView.check(matches(withText("Enter Street, Address, Concordia Classroom...")))
+
+        //Checks if Autocomplete options for text "h" are displayed
+        val searchAutoComplete = onView(
+            allOf(
+                withId(R.id.search_src_text), isDisplayed()
+            )
+        )
+        searchAutoComplete.perform(ViewActions.typeText("h"), ViewActions.closeSoftKeyboard())
+
+        Thread.sleep(2000)
+
+        //Checks if room suggestion H-400 is displayed and performs a click on the search card
+        val searchCardView = onView(
+            allOf(
+                withId(R.id.search_suggestions_card_view),
+                withChild(
+                    allOf(
+                        withId(R.id.relative_layout1),
+                        withChild(
+                            allOf(
+                                withId(R.id.search_suggestion),
+                                withText("H-400")
+                            )
+                        )
+                    )
+                ),
+                isDisplayed()
+            )
+        )
+
+        searchCardView.perform(click())
+
+        //Performs click on the Set Navigation Button
+        val setNavImageButton = onView(
+            allOf(
+                withId(R.id.setNavigationPoint),
+                withContentDescription("SetNavigation"),
+                hasSibling(
+                    allOf(
+                        withId(R.id.relative_layout1),
+                        withChild(
+                            allOf(
+                                withId(R.id.search_suggestion),
+                                withText("H-400")
+                            )
+                        )
+                    )
+                ),
+                isDisplayed()
+            )
+        )
+
+        setNavImageButton.perform(click())
+
+        //Checks if back button is displayed
+        val backImageButton = onView(
+            allOf(
+                withId(R.id.backFromNavigationButton), withContentDescription("Back"),
+                isDisplayed()
+            )
+        )
+        backImageButton.check(matches(isDisplayed()))
```

Comment:
As mentioned earlier, avoid declaring variable for onView method specially since it is not being reused.